### PR TITLE
Removes unused helper procs, slightly cleans up and sorts helper code

### PR DESCRIPTION
**Pull request by DominikPanic**
If it has `Carnie~` it's likely old

### CODE REVIEW

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -248,88 +242,42 @@ proc/listclearnulls(list/list)
...
-/proc/moveRange(list/L, fromIndex, toIndex, len=1)
+/proc/moveRange(list/L, fromIndex, toIndex, len = 1)
 	var/distance = abs(toIndex - fromIndex)
 	if(len >= distance)	//there are more elements to be moved than the distance to be moved. Therefore the same result can be achieved (with fewer operations) by moving elements between where we are and where we are going. The result being, our range we are moving is shifted left or right by dist elements
 		if(fromIndex <= toIndex)
 			return	//no need to move
 		fromIndex += len	//we want to shift left instead of right
 
-		for(var/i=0, i<distance, ++i)
+		for(var/i in 0 to distance)
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]: same here, it should probably be 1 to distance

File: code/__HELPERS/_lists.dm [hunk 2]
```diff
@@ -248,88 +242,42 @@ proc/listclearnulls(list/list)
...
 			L.Insert(fromIndex, null)
 			L.Swap(fromIndex, toIndex)
-			L.Cut(toIndex, toIndex+1)
+			L.Cut(toIndex, toIndex + 1)
 	else
 		if(fromIndex > toIndex)
 			fromIndex += len
 
-		for(var/i=0, i<len, ++i)
+		for(var/i in 0 to len)
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]: and here

File: code/__HELPERS/_lists.dm [hunk 3]
```diff
@@ -610,8 +455,8 @@ datum/proc/dd_SortValue()
 	if(!L)
 		return
 
-	for(var/i=1, i<L.len, ++i)
-		L.Swap(i,rand(i,L.len))
+	for(var/i in 1 to length(L))
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]: same here, 1 to length-1

File: code/__HELPERS/text.dm [hunk 4]
```diff
@@ -1,106 +1,109 @@
...
-			if(0 to 31)		return			//more weird stuff
-			if(32)			continue		//whitespace
-			else			non_whitespace = 1
-	if(non_whitespace)		return text		//only accepts the text if it has some non-spaces
+/proc/reject_bad_text(text, max_length = MAX_MESSAGE_LEN)
+	if(length(text) > max_length)	
+		return			//message too long
+
+	var/non_whitespace = FALSE
+	for(var/i in 1 to length(text))
```
Comment (spookydonut) [inline code/__HELPERS/text.dm]: same here, length-1

File: code/__HELPERS/_lists.dm [hunk 5]
```diff
@@ -1,167 +1,185 @@
...
 //Empties the list by setting the length to 0. Hopefully the elements get garbage collected
-proc/clearlist(list/list)
-	if(istype(list))
-		list.len = 0
-	return
+/proc/clearlist(list/L)
+	if(!istype(L))
+		return
+
+	L.len = 0
```
[the remark this answers is not in the document]
Comment (DominikPanic) [reply code/__HELPERS/_lists.dm]: You mean to profile it?

**Comment by DominikPanic** [comment]
Yeah I'll just revert all the for loop changes

File: code/__HELPERS/_lists.dm [hunk 6]
```diff
@@ -1,187 +1,208 @@
...
+		return
+
+	. = L[1]
+	L.Cut(1, 2)
+
 
 //Returns the next element in parameter list after first appearance of parameter element. If it is the last element of the list or not present in list, returns first element.
 /proc/next_in_list(element, list/L)
-	for(var/i=1, i<L.len, i++)
+	for(var/i = 1, i < length(L), i++)
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]:
```suggestion
	for(var/i in 1 to length(L)-1)
```

File: code/__HELPERS/_lists.dm [hunk 7]
```diff
@@ -1,187 +1,208 @@
...
 
 //Randomize: Return the list in a random order
 /proc/shuffle(list/L, ref) //Reference override for indexed lists.
 	if(!L)
 		return
 	L = L.Copy()
 
-	for(var/i=1, i<L.len, ++i)
-		L.Swap(i,rand(i,L.len))
+	for(var/i = 1, i < length(L), ++i)
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]:
```suggestion
	for(var/i in 1 to length(L)-1)
```

File: code/__HELPERS/_lists.dm [hunk 8]
```diff
@@ -248,88 +242,42 @@ proc/listclearnulls(list/list)
...
-/proc/moveRange(list/L, fromIndex, toIndex, len=1)
+/proc/moveRange(list/L, fromIndex, toIndex, len = 1)
 	var/distance = abs(toIndex - fromIndex)
 	if(len >= distance)	//there are more elements to be moved than the distance to be moved. Therefore the same result can be achieved (with fewer operations) by moving elements between where we are and where we are going. The result being, our range we are moving is shifted left or right by dist elements
 		if(fromIndex <= toIndex)
 			return	//no need to move
 		fromIndex += len	//we want to shift left instead of right
 
-		for(var/i=0, i<distance, ++i)
+		for(var/i = 0, i < distance, ++i)
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]:
```suggestion
		for(var/i in 1 to distance)
```

File: code/__HELPERS/_lists.dm [hunk 9]
```diff
@@ -248,88 +242,42 @@ proc/listclearnulls(list/list)
...
 			L.Insert(fromIndex, null)
 			L.Swap(fromIndex, toIndex)
-			L.Cut(toIndex, toIndex+1)
+			L.Cut(toIndex, toIndex + 1)
 	else
 		if(fromIndex > toIndex)
 			fromIndex += len
 
-		for(var/i=0, i<len, ++i)
+		for(var/i = 0, i < len, ++i)
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]:
```suggestion
		for(var/i in 1 to len)
```

File: code/__HELPERS/_lists.dm [hunk 10]
```diff
@@ -610,8 +455,8 @@ datum/proc/dd_SortValue()
 	if(!L)
 		return
 
-	for(var/i=1, i<L.len, ++i)
-		L.Swap(i,rand(i,L.len))
+	for(var/i = 1, i < length(L), ++i)
```
Comment (spookydonut) [inline code/__HELPERS/_lists.dm]:
```suggestion
	for(var/i in 1 to length(L)-1)
```

File: code/__HELPERS/sanitize_values.dm [hunk 11]
```diff
@@ -1,58 +1,82 @@
...
+	if(len != 7 && len !=4) 
+		return default
+
+	if(text2ascii(color,1) != 35) 
+		return default	//35 is the ascii code for "#"
+
 	. = "#"
-	for(var/i=2,i<=len,i++)
+
+	for(var/i = 2, i <= len, i++)
```
Comment (spookydonut) [inline code/__HELPERS/sanitize_values.dm]:
```suggestion
	for(var/i in 2 to len)
```

Comment (spookydonut) [reply code/__HELPERS/sanitize_values.dm]: if len is less than 2 nothing happens which is the same behaviour as before.

File: code/__HELPERS/text.dm [hunk 12]
```diff
@@ -1,106 +1,109 @@
...
-			if(0 to 31)		return			//more weird stuff
-			if(32)			continue		//whitespace
-			else			non_whitespace = 1
-	if(non_whitespace)		return text		//only accepts the text if it has some non-spaces
+/proc/reject_bad_text(text, max_length = MAX_MESSAGE_LEN)
+	if(length(text) > max_length)	
+		return			//message too long
+
+	var/non_whitespace = FALSE
+	for(var/i = 1, i <= length(text), i++)
```
Comment (spookydonut) [inline code/__HELPERS/text.dm]:
```suggestion
	for(var/i in 1 to length(text))
```

File: code/__HELPERS/text.dm [hunk 13]
```diff
@@ -1,106 +1,109 @@
...
+/proc/reject_bad_name(t_in, allow_numbers = FALSE, max_length = MAX_NAME_LEN)
 	if(!t_in || length(t_in) > max_length)
 		return //Rejects the input if it is null or if it is longer then the max length allowed
 
 	var/number_of_alphanumeric	= 0
 	var/last_char_group			= 0
 	var/t_out = ""
 
-	for(var/i=1, i<=length(t_in), i++)
+	for(var/i = 1, i <= length(t_in), i++)
```
Comment (spookydonut) [inline code/__HELPERS/text.dm]:
```suggestion
	for(var/i in 1 to length(t_in))
```

File: code/__HELPERS/type2type.dm [hunk 14]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+				. += A
+
+
+//Converts a bitfield to a list of numbers
+/proc/bitfield2list(bitfield = 0, list/L)
+	var/list/r = list()
+	if(islist(L))
+		var/max = min(length(L), 16)
+		var/bit = 1
+		for(var/i = 1, i <= max, i++)
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
		for(var/i in 0 to max-1)
```

File: code/__HELPERS/type2type.dm [hunk 15]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+			if(bitfield & bit)
+				r += L[i]
+			bit = bit << 1
+	else
+		for(var/bit = 1, bit <= 65535, bit = bit << 1)
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
		for(var/i in 0 to 23) // bitmasks are limited to 24bits
```

File: code/__HELPERS/type2type.dm [hunk 16]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+			if(bitfield & bit)
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
			if(bitfield & (1 << i))
```

File: code/__HELPERS/type2type.dm [hunk 17]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+				r += bit
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
				r += (1 << i)
```

File: code/__HELPERS/type2type.dm [hunk 18]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+			if(!findtext(value, "]"))
+				continue
+
+			value = splittext(value, "]") //"name]!" --> list("name","!")
+			for(var/A in value)
+				if(!var_source.vars.Find(A))
+					continue
+				. += A
+
+
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
#define MAX_BITFIELD_BITS 24
```

File: code/__HELPERS/type2type.dm [hunk 19]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+//Converts a bitfield to a list of numbers
+/proc/bitfield2list(bitfield = 0, list/L)
+	var/list/r = list()
+	if(islist(L))
+		var/max = min(length(L), 16)
+		var/bit = 1
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
```

File: code/__HELPERS/type2type.dm [hunk 20]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+				if(!var_source.vars.Find(A))
+					continue
+				. += A
+
+
+//Converts a bitfield to a list of numbers
+/proc/bitfield2list(bitfield = 0, list/L)
+	var/list/r = list()
+	if(islist(L))
+		var/max = min(length(L), 16)
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
		var/max = min(length(L), MAX_BITFIELD_BITS)
```

File: code/__HELPERS/type2type.dm [hunk 21]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+		var/bit = 1
+		for(var/i = 1, i <= max, i++)
+			if(bitfield & bit)
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
			if(bitfield & (1 << i))
```

File: code/__HELPERS/type2type.dm [hunk 22]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+				r += L[i]
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
				r += L[i+1]
```

File: code/__HELPERS/type2type.dm [hunk 23]
```diff
@@ -424,8 +438,29 @@ proc/tg_list2text(list/list, glue=",")
...
+			bit = bit << 1
```
Comment (spookydonut) [inline code/__HELPERS/type2type.dm]:
```suggestion
```

File: code/__HELPERS/unsorted.dm [hunk 24]
```diff
@@ -227,30 +160,29 @@ Turf and target are seperate in case you want to teleport some distance from a t
...
+	if(!findtext(key, "Guest-", 1, 7))
+		return FALSE
 
 	var/i = 7, ch, len = length(key)
 
 	if(copytext(key, 7, 8) == "W") //webclient
 		i++
 
-	for (, i <= len, ++i)
+	for(, i <= len, ++i)
```
Comment (spookydonut) [inline code/__HELPERS/unsorted.dm]:
```suggestion
	for(var/j in i to len)
```

File: code/__HELPERS/unsorted.dm [hunk 25]
```diff
@@ -227,30 +160,29 @@ Turf and target are seperate in case you want to teleport some distance from a t
...
 		ch = text2ascii(key, i)
```
Comment (spookydonut) [inline code/__HELPERS/unsorted.dm]:
```suggestion
		ch = text2ascii(key, j)
```

File: code/__HELPERS/text.dm [hunk 26]
```diff
@@ -284,56 +252,53 @@ proc/checkhtml(var/t)
...
-/proc/stringmerge(var/text,var/compare,replace = "*")
+/proc/stringmerge(text, compare, replace = "*")
 //This proc fills in all spaces with the "replace" var (* by default) with whatever
 //is in the other string at the same spot (assuming it is not a replace char).
 //This is used for fingerprints
 	var/newtext = text
 	if(lentext(text) != lentext(compare))
-		return 0
+		return FALSE
 	for(var/i = 1, i < lentext(text), i++)
```
Comment (spookydonut) [inline code/__HELPERS/text.dm]:
```suggestion
	for(var/i in 1 to lentext(text)-1)
```

File: code/__HELPERS/text.dm [hunk 27]
```diff
@@ -284,56 +252,53 @@ proc/checkhtml(var/t)
...
 
-/proc/stringpercent(var/text,character = "*")
+
+/proc/stringpercent(text, character = "*")
 //This proc returns the number of chars of the string that is the character
 //This is used for detective work to determine fingerprint completion.
 	if(!text || !character)
 		return 0
 	var/count = 0
 	for(var/i = 1, i <= lentext(text), i++)
```
Comment (spookydonut) [inline code/__HELPERS/text.dm]:
```suggestion
	for(var/i in 1 to lentext(text))
```

**Comment by Rohesie** [comment]
Merge conflict.

**Pull request merged**